pkg/controller/node/nodecontroller.go

```diff
@@ -371,6 +402,42 @@ func (nc *NodeController) Run() {
 			}
 		}, nc.nodeMonitorPeriod, wait.NeverStop)
 
+		go wait.Until(func() {
+			nc.evictorLock.Lock()
+			defer nc.evictorLock.Unlock()
+			for k := range nc.zoneIntolerablePodEvictor {
+				func(k string) {
+					nc.zoneIntolerablePodEvictor[k].Try(func(value TimedValue) (bool, time.Duration) {
+						message, _ := value.UID.(evictionMessage)
+						podName := message.podName
+						podNamespace := message.podNamespace
+						nodeUID := message.nodeUID
+
+						pod, err := nc.kubeClient.Core().Pods(podNamespace).Get(podName)
+						if err != nil {
+							// pod has been deleted, no more action needed
+							if apierrors.IsNotFound(err) {
+								return true, 0
+							}
+							glog.V(10).Infof("Failed to get pod %s/%s, err: %v.", podNamespace, podName, err)
+							return false, 0
+						}
+						EvictionsNumber.WithLabelValues(k).Inc()
+
+						remaining, err := deleteSinglePod(nc.kubeClient, nc.recorder, pod, string(nodeUID), nc.daemonSetStore)
+						if err != nil {
+							utilruntime.HandleError(fmt.Errorf("unable to evict pod %s: %v", value.Value, err))
```

Comment:
Wrong log, for now, but if you choose to apply changes I suggested above it'll be valid:)